### ISSUE TITLE
Open QR Code in new page on click

### ISSUE THREAD

**Issue by andrewtavis** [activist-org/activist#645]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/activist-org/activist/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow activist's [Code of Conduct](https://github.com/activist-org/activist/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

In #473 we were able to add in the ability to download QR Codes for pages in multiple formats. This functionality can be seen on say the [about page of an event](https://activist.org/en/events/123/about) where clicking the QR Code icon will open the modal where the download functionality is. It would be great if we could add in one final functionality to this modal, which would be that the user could click the QR Code and have it open in a new tab. We'd need to alert the user that this would happen by applying `cursor-pointer` to `QRCode` in [ModalQRCode](https://github.com/activist-org/activist/blob/main/frontend/components/modal/ModalQRCode.vue), and beyond this adding a tooltip on hover that indicates that clicking will open it in a new page would also be helpful.

### Contribution

Happy to support someone who wants to work on this!

**Comment by Vidhanvyrs** [comment]
hi @andrewtavis Can i try to work on this issue

**Comment by andrewtavis** [comment]
Hey @Vidhanvyrs! Nice to hear from you again 😊 Sure thing! Please let me know if I can do anything to help :)

**Comment by Vidhanvyrs** [comment]
hey @andrewtavis i need some help ! I just want to know how can i get the url of the qr code because my function is ready with the tooltip and all its just i was unable to find the url of the qrcode which we wanted to open in a new tab

**Comment by andrewtavis** [comment]
This is a great question, @Vidhanvyrs :) Made the issue and then realized that this could be more complex than I expected 🤔 I guess that we want to be able to open the image that's being saved in a new window, but there's no URL for it as it's generated in the frontend. Is there a way of copying the functionality of "open in a new window" without a URL? I guess not. We could make a URL for it that's based on the route for the event or organization and then at that route it'd just be the code? Can you think of anything else?

**Comment by Vidhanvyrs** [comment]
As we have a dynamically generated image on the frontend and we want that same image to open it in a new window, we can indeed create a temporary URL for it. One approach is to convert the image data into a data URL and use that as the source for the new window Now that is something to be researched iny give me some time may be I could come up with something that might solve it🙂

**Comment by andrewtavis** [comment]
Sounds wonderful, @Vidhanvyrs! We could also potentially make a route in `pages/events/[id]` and the like for organizations, if need be :) Let me know if there's something you find and want to check on!

**Comment by andrewtavis** [comment]
Hey @Vidhanvyrs 👋 How's all with this issue here? Let me know if there's anything I can do to support!

**Comment by Robot8lover** [comment]
Most modern browsers do not support opening a data URL in the top-frame (such as [Mozilla Firefox](https://blog.mozilla.org/security/2017/11/27/blocking-top-level-navigations-data-urls-firefox-59/) and [Google Chrome](https://groups.google.com/a/chromium.org/g/blink-dev/c/GbVcuwg_QjM). One approach is to open a new window using window.open, then write an image or iFrame to that window. This may be blocked by browsers as a pop-up window, and it is not as easy to save as an image since the page is not a simple image. This is the approach I used.

The better approach would probably be to, as was mentioned earlier here, create a backend version of the image that can be opened, potentially temporarily. This would allow for opening the actual image, which might be more intuitive. Also, it could allow for using an anchor element, which matches with its role better.

**Comment by andrewtavis** [comment]
Hey @Robot8lover! Thanks for the work here and further explanations! I'll review the PR soon - likely next week as I'm away this weekend :) 

Sounds like the current approach is fine for an initial take on this, and then I can compile the information in this issue into a new one to rework it based on a backend object. Will make that post review and tag you in there so you can keep an eye on it! 😊

**Comment by andrewtavis** [comment]
Closed by #887 🥳 Congrats on being contributor number 100, @Robot8lover! I had had it in mind for some time that it'd be good to give you the honor given all the work you picked up in one go :) :) Thanks again so much for this!

**Comment by Robot8lover** [comment]
@andrewtavis Thank you! It really means a lot. You have a very interesting project!

**Comment by andrewtavis** [comment]
Let's be in touch on another issue once everything else is merged 😊 There will definitely be some conflicts coming through as I've got loads of changes locally, but I'll handle them when they're up. Happy to also do a call at some point if you have questions about activist or want to talk the context of another issue!
